pre_commit/languages/php.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import unicode_literals
+
+import contextlib
+import json
+import os
+
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import Var
+from pre_commit.languages import helpers
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import resource_filename
+from pre_commit.xargs import xargs
+
+
+ENVIRONMENT_DIR = 'virtphpenv'
+VIRT_PHP_PATH = resource_filename('virtphp.phar')
+# It's currently impossible to move this file out of homedir
+VIRT_PHP_ENVIRONMENTS = os.path.join(
+    os.path.expanduser('~'),
+    '.virtphp',
+    'environments.json'
+)
+
+
+def get_env_patch(venv, language_version):
+    patches = (
+        ('VIRTPHP_COMPOSER_GLOBAL', '1'),
```

Comment:
tbh I have no idea if this is necessary. I mostly picked bits out of the virtphp bin/activate script that